nodebuilder/header/header.go

```diff
@@ -4,32 +4,78 @@ import (
 	"context"
 
 	"github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
 )
 
 // Module exposes the functionality needed for querying headers from the network.
 // Any method signature changed here needs to also be changed in the API struct.
 //
 //go:generate mockgen -destination=mocks/api.go -package=mocks . Module
 type Module interface {
+	// Head returns the ExtendedHeader of the chain head.
+	Head(context.Context) (*header.ExtendedHeader, error)
+
+	// Get returns the header of the given hash from the node's header store.
+	Get(ctx context.Context, hash libhead.Hash) (*header.ExtendedHeader, error)
+	// GetVerifiedRangeByHeight returns the given range [from:to) of ExtendedHeaders
+	// from the node's header store and verifies that the returned headers are
+	// adjacent to each other.
+	GetVerifiedRangeByHeight(
+		ctx context.Context,
+		from *header.ExtendedHeader,
+		to uint64,
+	) ([]*header.ExtendedHeader, error)
```

Comment:
I would omit Verified because our API should always return them verified. Our Getter interface will do this as well at some point

Reply:
We need to then discuss removing the method entirely from the interface (the non-verified range) 

cc @vgonkivs

Reply:
I'm happy to do this but in a subsequent PR pls @Wondertan so that it's an atomic change